### PR TITLE
ATL-15 Implement login page

### PR DESCRIPTION
**Pull request by korol1368**
TASK [ATL-15](https://alexsds.myjetbrains.com/youtrack/issue/ATL-15)

### CODE REVIEW

File: src/app/components/login/login.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import {LoginComponent} from './login.component';
+import {AuthService} from '@shared/services/auth.service';
+import {MatSnackBarModule} from '@angular/material/snack-bar';
+
+describe('LoginComponent', () => {
+  let component: LoginComponent;
+  let fixture: ComponentFixture<LoginComponent>;
+
+  beforeEach(async () => {
```
Comment (alexsds) [inline src/app/components/login/login.component.spec.ts:11]:
```
 beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [LoginComponent],
      providers: [AuthService],
      imports: [MatSnackBarModule, RouterTestingModule, FormsModule],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
    }).compileComponents();
  });
```

File: src/app/components/login/login.component.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      email: f.value.email,
+      password: f.value.password,
+    };
+
+    this.authService.login(user).subscribe((result) => {
+      if (result) {
+        f.reset();
+        this.router.navigate(['/list']);
+        this.snackBar.open('User authorized');
+      }
```
Comment (alexsds) [inline src/app/components/login/login.component.ts:37]:
```
} else {
      this.error = true;
}
```

File: src/app/components/login/login.component.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  onSubmit(f: NgForm): void {
+    if (f.invalid) {
+      return;
+    }
+
+    const user: User = {
+      email: f.value.email,
+      password: f.value.password,
+    };
+
```
Comment (alexsds) [inline src/app/components/login/login.component.ts:31]: this.error = false before call serice

File: src/app/components/login/login.component.spec.ts (hunk 1 above)
Comment (korol1368) [reply src/app/components/login/login.component.spec.ts:14]: Done

File: src/app/components/login/login.component.ts (hunk 2 above)
Comment (korol1368) [reply src/app/components/login/login.component.ts:41]: Done

File: src/app/components/login/login.component.ts (hunk 3 above)
Comment (korol1368) [reply src/app/components/login/login.component.ts:31]: Done

**Review by alexsds — APPROVED** [review]

**Pull request merged**